File: algorithms/fibonacci/fibonacci.py
def iterative_fibonacci(n):
    fibonacci = 1
    prev = 0
    prev_prev = None
    for x in range(1, n):
        prev_prev = prev
        prev = fibonacci
        fibonacci = prev + prev_prev

    return fibonacci


def recursive_fibonacci(n):
    if n <= 1:
        return n
    else:
        return recursive_fibonacci(n-1) + recursive_fibonacci(n-2)


f1 = iterative_fibonacci(16) 
f2 = recursive_fibonacci(16)
print(f1, f2)

File: algorithms/fibonacci/fibonacci.js
function iterativeFibonacci(n) {
  let fibonacci = 1;
  let prev = 0;
  let prevPrev = null;

  for (let i = 1; i < n; i++) {
    prevPrev = prev;
    prev = fibonacci;
    fibonacci = prev + prevPrev;
  }
  return fibonacci;
}

function recursiveFibonacci(n) {
  if (n <= 1) {
    return n;
  } else {
    return recursiveFibonacci(n - 1) + recursiveFibonacci(n - 2);
  }
}

const f1 = iterativeFibonacci(16);
const f2 = recursiveFibonacci(16);
console.log(f1, f2);
